ClassNotFoundException using quarkus-micrometer-registry-prometheus / quarkus-smallrye-graphql

Hi all, 

i hope its not a maven hickup on my side, I deleted my local repo and could reproduce the error.

## Describe the bug
Quarkus 1.12.2 Final using quarkus-micrometer-registry-prometheus:

```
<quarkus-plugin.version>1.12.2.Final</quarkus-plugin.version>
<quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
<quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
<quarkus.platform.version>1.12.2.Final</quarkus.platform.version>
```

```
<dependency>
<groupId>io.quarkus</groupId>
<artifactId>quarkus-micrometer-registry-prometheus</artifactId>
</dependency>
```
maybe in conjunction with  : 

```
<dependency>
<groupId>io.quarkus</groupId>
<artifactId>quarkus-smallrye-graphql</artifactId>
</dependency>
```
@phillip-kruger (since I think you are working on a lot of graphql matters and im not sure about if this is a graphql issue)?

I get the following exception:

```
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/Tag
	at io.smallrye.graphql.cdi.metrics.MetricsService.getTags(MetricsService.java:84)
	at io.smallrye.graphql.cdi.metrics.MetricsService.createOperation(MetricsService.java:37)
	at io.smallrye.graphql.execution.event.EventEmitter.fireCreateOperation(EventEmitter.java:142)
	at io.smallrye.graphql.bootstrap.Bootstrap.addRootObject(Bootstrap.java:180)
	at io.smallrye.graphql.bootstrap.Bootstrap.addQueries(Bootstrap.java:148)
	at io.smallrye.graphql.bootstrap.Bootstrap.generateGraphQLSchema(Bootstrap.java:122)
	at io.smallrye.graphql.bootstrap.Bootstrap.bootstrap(Bootstrap.java:98)
	at io.smallrye.graphql.cdi.producer.GraphQLProducer.initialize(GraphQLProducer.java:44)
```

### Expected behavior
Quarkus starts normally.

### Actual behavior
See exception.

adding the following dependency solves the problem for me:

```
<dependency>
<groupId>org.eclipse.microprofile.metrics</groupId>
<artifactId>microprofile-metrics-api</artifactId>
</dependency>
```

I would expect to either include the metrics reference in the tutorial if actually needed, or add it directly to the 
_quarkus-micrometer-registry-prometheus_ and/or others which maybe need the reference.

Hope someone can reproduce this issue.

Best

Jan

--- a/extensions/smallrye-graphql/runtime/pom.xml
+++ b/extensions/smallrye-graphql/runtime/pom.xml
@@ -39,6 +39,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-smallrye-context-propagation</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.metrics</groupId>
+            <artifactId>microprofile-metrics-api</artifactId>
+        </dependency>
     </dependencies>
 
     <build>

